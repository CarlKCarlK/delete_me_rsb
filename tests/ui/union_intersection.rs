use range_set_int::{intersection, union};

fn main() {
    let _u = union([[(1, 3), (2, 4)].into_iter()]);
    let _i = intersection([[(1, 3), (2, 4)].into_iter()]);
}
