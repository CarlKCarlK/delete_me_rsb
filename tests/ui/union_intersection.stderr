error[E0277]: the trait bound `std::array::IntoIter<({integer}, {integer}), 2>: SortedDisjoint` is not satisfied
 --> tests/ui/union_intersection.rs:4:14
  |
4 |     let _u = union([[(1, 3), (2, 4)].into_iter()]);
  |              ^^^^^ the trait `SortedDisjoint` is not implemented for `std::array::IntoIter<({integer}, {integer}), 2>`
  |
  = help: the following other types implement trait `SortedDisjoint`:
            BitOrIter<T, I>
            DynSortedDisjoint<'a, T>
            NotIter<T, I>
            Ranges<'_, T>
            itertools::tee::Tee<I>
            range_set_int::unsorted_disjoint::SortedDisjointWithLenSoFar<T, I>
note: required by a bound in `union`
 --> src/lib.rs
  |
  |     J: Iterator<Item = (T, T)> + SortedDisjoint,
  |                                  ^^^^^^^^^^^^^^ required by this bound in `union`

error[E0277]: the trait bound `std::array::IntoIter<({integer}, {integer}), 2>: SortedStarts` is not satisfied
 --> tests/ui/union_intersection.rs:4:14
  |
4 |     let _u = union([[(1, 3), (2, 4)].into_iter()]);
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `SortedStarts` is not implemented for `std::array::IntoIter<({integer}, {integer}), 2>`
  |
  = help: the following other types implement trait `SortedStarts`:
            BitOrIter<T, I>
            DynSortedDisjoint<'a, T>
            NotIter<T, I>
            Ranges<'_, T>
            itertools::adaptors::MergeBy<L, R, for<'a, 'b> fn(&'a (T, T), &'b (T, T)) -> bool>
            itertools::kmerge_impl::KMergeBy<I, for<'a, 'b> fn(&'a (T, T), &'b (T, T)) -> bool>
            itertools::tee::Tee<I>
            range_set_int::unsorted_disjoint::AssumeSortedStarts<T, I>
          and $N others
  = note: required for `itertools::kmerge_impl::KMergeBy<std::array::IntoIter<({integer}, {integer}), 2>, ...>` to implement `SortedStarts`
note: required by a bound in `BitOrIter`
 --> src/lib.rs
  |
  |     I: Iterator<Item = (T, T)> + SortedStarts,
  |                                  ^^^^^^^^^^^^ required by this bound in `BitOrIter`

error[E0277]: the trait bound `std::array::IntoIter<({integer}, {integer}), 2>: SortedDisjoint` is not satisfied
 --> tests/ui/union_intersection.rs:5:14
  |
5 |     let _i = intersection([[(1, 3), (2, 4)].into_iter()]);
  |              ^^^^^^^^^^^^ the trait `SortedDisjoint` is not implemented for `std::array::IntoIter<({integer}, {integer}), 2>`
  |
  = help: the following other types implement trait `SortedDisjoint`:
            BitOrIter<T, I>
            DynSortedDisjoint<'a, T>
            NotIter<T, I>
            Ranges<'_, T>
            itertools::tee::Tee<I>
            range_set_int::unsorted_disjoint::SortedDisjointWithLenSoFar<T, I>
note: required by a bound in `intersection`
 --> src/lib.rs
  |
  |     J: Iterator<Item = (T, T)> + SortedDisjoint,
  |                                  ^^^^^^^^^^^^^^ required by this bound in `intersection`

error[E0277]: the trait bound `std::array::IntoIter<({integer}, {integer}), 2>: SortedDisjoint` is not satisfied
 --> tests/ui/union_intersection.rs:5:14
  |
5 |     let _i = intersection([[(1, 3), (2, 4)].into_iter()]);
  |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `SortedDisjoint` is not implemented for `std::array::IntoIter<({integer}, {integer}), 2>`
  |
  = help: the following other types implement trait `SortedDisjoint`:
            BitOrIter<T, I>
            DynSortedDisjoint<'a, T>
            NotIter<T, I>
            Ranges<'_, T>
            itertools::tee::Tee<I>
            range_set_int::unsorted_disjoint::SortedDisjointWithLenSoFar<T, I>
note: required by a bound in `NotIter`
 --> src/lib.rs
  |
  |     I: Iterator<Item = (T, T)> + SortedDisjoint,
  |                                  ^^^^^^^^^^^^^^ required by this bound in `NotIter`
